### PR TITLE
Add the ability to have dynamically built Odata schemas

### PR DESCRIPTION
**Pull request by grantspeelman**
Add the ability to dynamically build schema.

See README for what this makes possible : https://github.com/NEXL-LTS/odata_duty-ruby/pull/2/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R80

### CODE REVIEW

File: lib/odata_duty.rb [hunk 1]
```diff
@@ -253,91 +218,16 @@ def self.index_hash(metadata_url)
...
-      end
-    end
-
-    def self.add_next_link(data, endpoint, set_builder, query_options, context)
-      return unless set_builder.od_next_link_skiptoken
-
-      next_query_options = query_options.merge(url: endpoint.url,
-                                               :$skiptoken => set_builder.od_next_link_skiptoken)
-      data[:'@odata.nextLink'] = context.url_for(**next_query_options)
+      Executor.execute(url: url, context: context, query_options: query_options, schema: self)
```
Comment (grantspeelman) [inline lib/odata_duty.rb:230]: moved a lot of the logic into Executor class, so the same execution logic can work for static and dynamic schemas.

**Review by grdmnt — APPROVED** [review]
Looks good! Took some time to understand how it works, but I think I got the gist of the whole feature. I think it's something that we can talk about during backend chat.

File: lib/odata_duty/edms.rb [hunk 2]
```diff
@@ -54,6 +86,14 @@ def self.property_type
       'Edm.DateTimeOffset'
     end
 
+    OAS_TYPE = { 'type' => 'string', 'format' => 'date-time' }.freeze
```
Comment (bapu-nexl) [inline lib/odata_duty/edms.rb:89]: Did you mean to define it twice? Wouldn't there be only one defination after the file is loaded?

**Review by bapu-nexl — APPROVED** [review]
Looks good

File: lib/odata_duty/edms.rb (hunk 2 above)
Comment (grantspeelman) [reply lib/odata_duty/edms.rb:89]: can see from here, but each constant is inside a different class.

**Pull request merged**